lua/websearcher.lua

```diff
@@ -1,60 +1,259 @@
 local M = {}
 
 local config = {
-	open_cmd = "",
-	search_engine = "Google",
+    open_cmd = "",
+    search_engine = "Google",
+    use_w3m = false,
 }
 
 local SearchEngineURLs = {
-	Google = "https://google.com/search?q=",
-	DuckDuckGo = "https://duckduckgo.com/?q=",
-	Bing = "https://www.bing.com/search?q=",
+    Google = "https://google.com/search?q=",
+    DuckDuckGo = "https://duckduckgo.com/?q=",
+    Bing = "https://www.bing.com/search?q=",
+    Yahoo = "https://search.yahoo.com/search?p=",
+    Baidu = "https://www.baidu.com/s?wd=",
+    Yandex = "https://yandex.com/search/?text=",
+    Ask = "https://www.ask.com/web?q=",
+    Ecosia = "https://www.ecosia.org/search?q=",
+    Phind = "https://www.phind.com/search?q=",
+    Wikipedia = "https://en.wikipedia.org/w/index.php?search=",
 }
 
+local search_history = {}
+
+--- Save search to history
+---@param query string
+local function save_to_history(query)
```

Comment:
When I was developing the PR and testing. I saw it as a utility that could help me to be able to keep track.

While it is completely browser independent, I think it is an added bonus that you can check from the editor the last searches made with WebSearcher.

Reply:
> I think it is an added bonus that you can check from the editor the last searches made with WebSearcher.

I see, so that's another way of thinking about it. 🤔
I'll respect your opinion—thank you for offering a fresh perspective.
I'll consider this further moving forward.